integration/tests/cook/test_cli.py

```diff
@@ -398,3 +398,88 @@ def test_complex_commands(self):
+        waiting_uuid = uuids[0]
+        # running
+        cp, uuids = cli.submit('sleep 60', self.cook_url, submit_flags='--name %s' % name)
+        self.assertEqual(0, cp.returncode, cp.stderr)
+        util.wait_for_job(self.cook_url, uuids[0], 'running')
+        cp, jobs = self.list_jobs(name, user, 'running')
+        self.assertEqual(0, cp.returncode, cp.stderr)
+        self.assertEqual(1, len(jobs))
+        self.assertEqual(uuids[0], jobs[0]['uuid'])
+        running_uuid = uuids[0]
+        # completed
+        cp, uuids = cli.submit('ls', self.cook_url, submit_flags='--name %s' % name)
+        self.assertEqual(0, cp.returncode, cp.stderr)
+        util.wait_for_job(self.cook_url, uuids[0], 'completed')
+        cp, jobs = self.list_jobs(name, user, 'completed')
+        self.assertEqual(0, cp.returncode, cp.stderr)
+        self.assertEqual(1, len(jobs))
+        self.assertEqual(uuids[0], jobs[0]['uuid'])
+        # success
+        cp, jobs = self.list_jobs(name, user, 'success')
+        self.assertEqual(0, cp.returncode, cp.stderr)
+        self.assertEqual(1, len(jobs))
+        self.assertEqual(uuids[0], jobs[0]['uuid'])
+        success_uuid = uuids[0]
+        # failed
+        cp, uuids = cli.submit('exit 1', self.cook_url, submit_flags='--name %s' % name)
+        self.assertEqual(0, cp.returncode, cp.stderr)
+        util.wait_for_job(self.cook_url, uuids[0], 'completed')
+        cp, jobs = self.list_jobs(name, user, 'failed')
+        self.assertEqual(0, cp.returncode, cp.stderr)
+        self.assertEqual(1, len(jobs))
+        self.assertEqual(uuids[0], jobs[0]['uuid'])
+        failed_uuid = uuids[0]
+        # all
+        cp, jobs = self.list_jobs(name, user, 'all')
+        uuids = [j['uuid'] for j in jobs]
+        self.assertEqual(0, cp.returncode, cp.stderr)
+        self.assertEqual(4, len(jobs))
+        self.assertIn(waiting_uuid, uuids)
+        self.assertIn(running_uuid, uuids)
+        self.assertIn(success_uuid, uuids)
+        self.assertIn(failed_uuid, uuids)
+        # waiting+running
+        cp, jobs = self.list_jobs(name, user, 'waiting+running')
+        uuids = [j['uuid'] for j in jobs]
+        self.assertEqual(0, cp.returncode, cp.stderr)
+        self.assertEqual(2, len(jobs))
+        self.assertIn(waiting_uuid, uuids)
+        self.assertIn(running_uuid, uuids)
+        # completed+waiting
+        cp, jobs = self.list_jobs(name, user, 'completed+waiting')
+        uuids = [j['uuid'] for j in jobs]
+        self.assertEqual(0, cp.returncode, cp.stderr)
+        self.assertEqual(3, len(jobs))
+        self.assertIn(waiting_uuid, uuids)
+        self.assertIn(success_uuid, uuids)
+        self.assertIn(failed_uuid, uuids)
+
+    def test_list_invalid_state(self):
+        cp = cli.list_jobs(self.cook_url, '--state foo')
+        self.assertEqual(2, cp.returncode, cp.stderr)
+        self.assertIn('foo is not a valid state filter', cli.decode(cp.stderr))
```

Comment:
Add test for bad state filter component, e.g.: `running+foo`

Reply:
No longer necessary since it's now a repeated argument